packages/vertexai/src/requests/schema-builder.ts

```diff
@@ -0,0 +1,197 @@
+import { SchemaInterface, SchemaType, SchemaParams } from '../types/schema';
+
+export abstract class Schema implements SchemaInterface {
+  /**
+   * Optional. The type of the property. {@link
+   * SchemaType}.
+   */
+  type: SchemaType;
+  /** Optional. The format of the property.
+   * Supported formats:
+   *  for NUMBER type: "float", "double"
+   *  for INTEGER type: "int32", "int64"
+   *  for STRING type: "email", "byte", etc
+   */
+  format?: string;
+  /** Optional. The description of the property. */
+  description?: string;
+  /** Optional. Whether the property is nullable. Defaults to false. */
+  nullable: boolean;
+  /** Optional. Defaults to true. */
+  required: boolean;
+  /** Optional. The example of the property. */
+  example?: unknown;
+
+  constructor(schemaParams: SchemaInterface) {
+    this.type = schemaParams.type;
+    this.format = schemaParams?.format;
+    this.description = schemaParams?.description;
+    this.example = schemaParams?.example;
+    this.nullable = schemaParams.hasOwnProperty('nullable')
+      ? !!schemaParams.nullable
+      : false;
+    this.required = schemaParams.hasOwnProperty('required')
+      ? !!schemaParams.required
+      : true;
+  }
+
+  /** Converts class to a plain JSON object (not a string). */
+  toJSON(): Record<string, unknown> {
+    const obj: Record<string, unknown> = {};
+    for (const prop in this) {
+      if (this.hasOwnProperty(prop) && this[prop] !== undefined) {
```

Comment:
TS/ESlint yells about that, I think it's because classes have some built-in inherited native properties and hasOwnProperty skips them and focuses on the properties you explicitly put on it.